nncf/experimental/tensorflow/graph/argprovider.py

```diff
@@ -0,0 +1,229 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from nncf.common.utils.registry import Registry
+
+
+def replace(xs, pos: int, value):
+    """
+    Return a new instance of the tuple replacing the specified
+    position with the new value.
+
+    :param xs: A tuple.
+    :param pos: Zero-based index of the item which should be replaced.
+    :param value: New value.
+    """
+    return tuple(value if idx == pos else elem for idx, elem in enumerate(xs))
+
+
+def check_port_id(port_id: int, min_port_id: int, max_port_id: int):
+    if min_port_id <= port_id <= max_port_id:
+        return
+    raise ValueError(f'Unexpected `port_id`: {port_id}')
+
+
+TF_ARG_PROVIDERS = Registry('TF_ARG_PROVIDERS')
+
+
+class ArgProvider:
+    """
+    Base class for all argument providers. An `ArgProvider` instance
+    describes how to extract input or output argument with specified
+    `port_id` from the `args` and `kwargs`.
+    """
+
+    def get_input(self, input_port_id: int, args, kwargs):
+        """
+        Returns input Tensor with specified `input_port_id`.
+        :param input_port_id: Zero-based index an input Tensor
+            for TensorFlow operation.
+        :return: A Tensor with specified `input_port_id`.
+        """
+
+    def get_output(self, output_port_id: int, args, kwargs):
+        """
+        Returns output Tensor with specified `output_port_id`.
+        :param output_port_id: Zero-based index an output Tensor
+            of TensorFlow operation.
+        :return: A Tensor with specified `output_port_id`.
+        """
```

Comment:
Return type hints are missing for the getters, value type hints are missing for setters.

Reply:
Thanks for the comment! I fixed it.